Mono.Addins/Mono.Addins.Database/AddinFolderVisitor.cs

```diff
@@ -66,7 +67,7 @@ protected virtual void OnVisitFolder (IProgressStatus monitor, string path, stri
 			if (!FileSystem.DirectoryExists (path))
 				return;
 			
-			var files = FileSystem.GetFiles (path);
+			var files = FileSystem.GetFiles (path).ToArray();
```

Comment:
files is ToArray()'d here because the enumerable is used 3 times - which could end up hitting the disk 3 times.